components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/request/CASIdentityRequestFactory.java

```diff
@@ -52,7 +52,8 @@ public String getName() {
     public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
         String serviceProviderUrl = request.getParameter(CASConstants.CASSSOConstants.SERVICE_PROVIDER_ARGUMENT);
         String ticket = request.getParameter(CASConstants.CASSSOConstants.SERVICE_TICKET_ARGUMENT);
-        return StringUtils.isNotBlank(serviceProviderUrl) || StringUtils.isNotBlank(ticket);
+        boolean logout = request.getRequestURI().contains("logout");
```

Comment:
Move the "logout" constant to the CASConstants.java and import it from the constants file